### ISSUE TITLE
try-except-raise does not consider tuples

### ISSUE THREAD

**Issue by NotSqrt** [PyCQA/pylint#2389]
This issue is very close to #2284, but handling of tuples does not seem to be considered.

### Steps to reproduce

```python

def call():
    pass


def handler():
    try:
        call()
    except (FileNotFoundError, PermissionError):
        raise
    except OSError:
        print("specific treatment for other OSError")
        raise


def handler2():
    # Fixed by https://github.com/PyCQA/pylint/issues/2284
    try:
        call()
    except FileNotFoundError:
        raise
    except OSError:
        print("specific treatment for other OSError")
        raise
```

### Current behavior

```
test_pylint_try_except.py:9:4: W0706: The except handler raises immediately (try-except-raise)
```

### Expected behavior

No issue should be reported because we catch some subclasses of `OSError` (`(FileNotFoundError, PermissionError)`)  explicitly before the more general `except OSError`.

### pylint --version output

```
pylint 2.1.1
astroid 2.0.1
Python 3.6.2 (default, Jul 17 2017, 23:14:31) 
[GCC 5.4.0 20160609]
```

**Comment by sushobhit27** [comment]
Voila!!! how many test cases exists for this checker :cry: :sob: :rofl: :laughing:  I will fix it soon.

**Comment by NotSqrt** [comment]
Hello, real world !

**Comment by sushobhit27** [comment]
I have considered following use cases (consider tuples in non bare-raise except handler also):
```
try:
    pass
except (FileNotFoundError, PermissionError):
    raise
except (OverflowError, OSError):
    print("a failure")

try:
    pass
except (FileNotFoundError, PermissionError):  # [try-except-raise]
    raise
except (OverflowError, ZeroDivisionError):
    print("a failure")
```
